file-store/src/main/java/org/keycloak/models/map/storage/file/MapEntityYamlContext.java

```diff
@@ -151,16 +151,47 @@ public YamlContext<?> getContext(String nameOfSubcontext) {
         if (ef != null) {
             if (ef.getCollectionElementClass() != Void.class) {
                 return new DefaultListContext();
-            } else if (ef.getMapValueClass() == List.class) {
+            } else if (ef.getMapValueClass() != Void.class) {
                 return new DefaultMapContext();
             }
-            return null;
+            return new DefaultObjectContext();
         }
         LOG.warnf("No special context set for field %s", nameOfSubcontext);
         return null;
     }
 
-    private static class MapEntitySequenceYamlContext<T> extends DefaultListContext {
+    @Override
+    public void writeValue(T entity, WritingMechanism mech, Runnable doNothing) {
```

Comment:
The name `doNothing` is misleading. The runnable must be executed, just like it used to be, and it must be executed only upon first write.